drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java

```diff
@@ -0,0 +1,1042 @@
+                                                return;
+                                            }
+                                            String cm = GuidedDecisionTableConstants.INSTANCE.DeleteConditionColumnWarning0( c.getHeader() );
+                                            if ( Window.confirm( cm ) ) {
+                                                presenter.getActiveDecisionTable().deleteColumn( (LimitedEntryBRLConditionColumn) c );
+                                            }
+                                        }
+                                    } );
+
+        } else if ( c instanceof BRLConditionColumn ) {
+            return new ImageButton( image,
+                                    GuidedDecisionTableConstants.INSTANCE.RemoveThisConditionColumn(),
+                                    new ClickHandler() {
+                                        public void onClick( ClickEvent w ) {
+                                            if ( !presenter.getActiveDecisionTable().canConditionBeDeleted( (BRLConditionColumn) c ) ) {
+                                                Window.alert( GuidedDecisionTableConstants.INSTANCE.UnableToDeleteConditionColumn0( c.getHeader() ) );
+                                                return;
+                                            }
+                                            String cm = GuidedDecisionTableConstants.INSTANCE.DeleteConditionColumnWarning0( c.getHeader() );
+                                            if ( Window.confirm( cm ) ) {
+                                                presenter.getActiveDecisionTable().deleteColumn( (BRLConditionColumn) c );
+                                            }
+                                        }
+                                    } );
+
+        }
+        return new ImageButton( image,
+                                GuidedDecisionTableConstants.INSTANCE.RemoveThisConditionColumn(),
+                                new ClickHandler() {
+                                    public void onClick( ClickEvent w ) {
+                                        if ( !presenter.getActiveDecisionTable().canConditionBeDeleted( c ) ) {
+                                            Window.alert( GuidedDecisionTableConstants.INSTANCE.UnableToDeleteConditionColumn0( c.getHeader() ) );
+                                            return;
+                                        }
+                                        String cm = GuidedDecisionTableConstants.INSTANCE.DeleteConditionColumnWarning0( c.getHeader() );
+                                        if ( Window.confirm( cm ) ) {
+                                            presenter.getActiveDecisionTable().deleteColumn( c );
+                                        }
+                                    }
+                                } );
```

Comment:
These 3 ImageButton blocks are the same except for the type cast... Could the code be reused instead of duplicating?